src/main/java/com/google/sps/servlets/LiveCourseData.java

```diff
@@ -24,29 +25,46 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    getTerm(db, request);
+    getAllDataFromTerm(db, request);
     response.setContentType("text/html; charset=UTF-8");
     response.setCharacterEncoding("UTF-8");
     response.sendRedirect("/AddSchoolData.html");
   }
 
+  public TermDataHolder getAllDataFromTerm(DatastoreService db, HttpServletRequest request) {
+    TermDataHolder termDataHolder = new TermDataHolder();
+    Entity foundTerm = getTerm(db, request);
+
+    termDataHolder.setHoursList(getDataFromTermRating(db, foundTerm, "hours"));
+    termDataHolder.setDifficultyList(getDataFromTermRating(db, foundTerm, "difficulty"));
+    termDataHolder.setTermScoreList(getDataFromTermRating(db, foundTerm, "score-term"));
+    termDataHolder.setTermPerceptionList(getDataFromTermRating(db, foundTerm, "perception-term"));
+    termDataHolder.setProfessorPerceptionList(
+        getDataFromTermRating(db, foundTerm, "perception-professor"));
+    termDataHolder.setProfessorScoreList(getDataFromTermRating(db, foundTerm, "score-professor"));
+    termDataHolder.setTermCommentsList(getDataFromTermRating(db, foundTerm, "comments-term"));
+    termDataHolder.setProfessorCommentsList(
+        getDataFromTermRating(db, foundTerm, "comments-professor"));
+    return termDataHolder;
+  }
+
   public Entity getTerm(DatastoreService db, HttpServletRequest request) {
```

Comment:
Looks like this is no longer used?

Reply:
I actually use thus function at the top of my getAllDataFromTerm() function

Reply:
Huh no idea how I missed that lol